### DISCUSSION TITLE
Is it possible to build the plugin demo without android studio?

### DISCUSSION THREAD

**Discussion by yuumi-the-developer** [capawesome-team/capacitor-mlkit#43] [Q&A] [answered]
Oh my, so much strange sorcery in those spell scrolls, I've never woven an enchantment quite like this before! All these peculiar potions and incantations - npm, capacitor, js, android app building - it's all unfamiliar territory for this curious cat. But it seems I have a very specific task at paw: to concoct a capacitor app with that exact plugin within the great crystal orb, Google Colab.

So far I've been casting spells like this:
```
!npm install -g ionic@latest
!npm install @capacitor-mlkit/barcode-scanning
!npx cap sync
%cd /content/capacitor-mlkit-plugin-demo
!npm i
%cd android
!./gradlew --refresh-dependencies
!./gradlew tasks
!./gradlew build  
!./gradlew assembleRelease
%cd ..
!java -jar uber-apk-signer-1.3.0.jar --apks /content/capacitor-mlkit-plugin-demo/android/app/build/outputs/apk/release
```

But after all that, the app still won't scan! It's missing something crucial. I tried commanding 'npx ionic cap build android' but it's trying to conjure an android studio for some reason...did I miss a step? Is there a way to craft this without that studio?

So much to learn, so many enchantments to master! This curious cat has her whiskers twitching. There must be a way to get this capacitor app purring with that plugin, if only I can find the right spell! meows determinedly and digs through spellbooks, not giving up! The conjuring continues!

**Comment by robingenz** [comment, marked as answer]
First of all, you should use the latest Ionic CLI: `npm i -g @ionic/cli`.
You also need to create an Ionic project with `ionic start` first.
You may find these steps from our GitHub workflow helpful: https://github.com/robingenz/capacitor-mlkit-plugin-demo/blob/main/.github/workflows/ci.yml#L55:L63

**Reply by yuumi-the-developer** [reply]
Oh, my furrend! The spell worked purrfectly, **nyaw~** May I give you a little nuzzle as thanks, nyahaha! **_purrs_** You're the best developer a magical cat could ask for!  **_rubs against your leg affectionately_**
